src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -111,4 +128,109 @@ private void addListingToBusinesses(PlacesSearchResult place) {
+                                .setApplicationName("linkedinSearch") 
+                                .setGoogleClientRequestInitializer(
+                                      new CustomsearchRequestInitializer(KEY)) 
+                                .build();
+
+    Customsearch.Cse.List list = cs.cse().list(currentBusinessName).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    int companyFollowers = 0;
+    if (searchJsonResults!=null && searchJsonResults.size() != 0) {
+      Result linkedinBusiness = searchJsonResults.get(0);
+      String businessDescription = (String) linkedinBusiness.getPagemap()
+                                                            .get("metatags")
+                                                            .get(0)
+                                                            .get("og:description");
+      if(businessDescription.indexOf(startIndex) != -1 
+          && businessDescription.indexOf(endIndex) != -1){
+        String followers = businessDescription.substring(
+                                businessDescription.indexOf(startIndex) + 2, 
+                                businessDescription.indexOf(endIndex) - 1);
+        companyFollowers = 
+                    Integer.parseInt(followers.replaceAll(",", ""));
+      }
+      if (companyFollowers > minFollowers) {
+        addBigBusinessToDatabase();
+      } else {
+        checkNumberOfSimilarBusinessesInTheArea(currentBusiness.getName(), similarBusinessesInTheArea);
+      }
+    }
+  }
+
+  private void checkNumberOfSimilarBusinessesInTheArea(String businessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) {
+    int countNumberOfMatchingBusiness = 0;
+    int i = 0;
+    while (i < similarBusinessesInTheArea.length 
+          && countNumberOfMatchingBusiness < 10) {
+      if(similarBusinessesInTheArea[i].vicinity != null){
+        if (similarBusinessesInTheArea[i].name.contains(businessName) 
+            && !(similarBusinessesInTheArea[i].vicinity.equals(currentBusiness.getFormattedAddress()))) {
+          countNumberOfMatchingBusiness++;
+        }
+      }
+      i++;
+     }
+     if (countNumberOfMatchingBusiness >= 10) {
+       addBigBusinessToDatabase();
+     }
+   }
+
+  private void addBigBusinessToDatabase(){
+    businesses.remove();
+    String title = "Business";
+    String businessTypes = "BusinessTypes";
+    String address = "Address";
+    String rating = "Rating";
+    String photos = "Photos";
+    Entity businessEntity = new Entity("BigBusinesses");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    businessEntity.setProperty(title, currentBusiness.getName());
+    businessEntity.setProperty(address, currentBusiness.getFormattedAddress());
+    businessEntity.setProperty(rating, currentBusiness.getRating());
+    businessEntity.setProperty(businessTypes, 
```

Comment:
We definitely want to make sure that the url gets added to the datastore properties -- going to make an issue to track this. It depends which PR gets approved first.

Reply:
Sounds good